src/Tribe/Repositories/Traits/Post_Attendees.php

```diff
@@ -0,0 +1,271 @@
+
+		if ( $has_attendees ) {
+			// Join to the meta that relates attendees to events.
+			$repo->filter_query->join( "
+					LEFT JOIN `{$wpdb->postmeta}` AS `{$alias_event}`
+						ON `{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+				", $alias_event );
+
+			// Join to the meta that relates meta to attendees.
+			$repo->filter_query->join( "
+					LEFT JOIN `{$wpdb->posts}` AS `{$alias_attendee}`
+						ON `{$alias_attendee}`.`ID` = `{$alias_event}`.`post_id`
+				", $alias_attendee );
+
+			$repo->where_clause( "
+					`{$alias_event}`.`meta_key` IN ( {$event_meta_keys} )
+					AND `{$alias_attendee}`.`post_type` IN ( {$attendee_types} )
+				" );
+
+			return;
+		}
+
+		// Handle case where post has no attendees.
+		$repo->where_clause( "
+				NOT EXISTS (
+					SELECT 1
+					FROM
+						`{$wpdb->postmeta}` AS `{$alias_event}`
+					LEFT JOIN `{$wpdb->posts}` AS `{$alias_attendee}`
+						ON `{$alias_attendee}`.`ID` = `{$alias_event}`.`post_id`
+					WHERE
+						`{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+						AND `{$alias_event}`.`meta_key` IN ( {$event_meta_keys} )
+						AND `{$alias_attendee}`.`post_type` IN ( {$attendee_types} )
+					LIMIT 1
+				)
+			" );
+	}
+
+	/**
+	 * Filters events to include only those that do not match the provided attendee(s).
+	 *
+	 * @since TBD
+	 *
+	 * @param int|array $attendee_ids The attendee(s) to filter out.
+	 */
+	public function filter_by_attendee_not_in( $attendee_ids ) {
+		$repo = $this;
+
+		// If the repo is decorated, use that.
+		if ( ! empty( $this->decorated ) ) {
+			$repo = $this->decorated;
+		}
+
+		global $wpdb;
+
+		$attendee_ids = (array) $attendee_ids;
+		$attendee_ids = array_map( 'absint', $attendee_ids );
+		$attendee_ids = array_unique( $attendee_ids );
+		$attendee_ids = implode( ', ', $attendee_ids );
+
+		$alias_event    = 'sub_ticket_attendee_event';
+		$alias_attendee = 'sub_ticket_attendee';
+
+		$event_meta_keys = method_exists( $this, 'attendee_to_event_keys' ) ? $this->attendee_to_event_keys() : [];
+		$event_meta_keys = array_map( [ $wpdb, '_real_escape' ], $event_meta_keys );
+		$event_meta_keys = '"' . implode( '", "', $event_meta_keys ) . '"';
+
+		$attendee_types = method_exists( $this, 'attendee_types' ) ? $this->attendee_types() : [];
+		$attendee_types = array_map( [ $wpdb, '_real_escape' ], $attendee_types );
+		$attendee_types = '"' . implode( '", "', $attendee_types ) . '"';
+
+		$repo->where_clause( "
+				NOT EXISTS (
+					SELECT 1
+					FROM
+						`{$wpdb->postmeta}` AS `{$alias_event}`
+					LEFT JOIN `{$wpdb->posts}` AS `{$alias_attendee}`
+						ON `{$alias_attendee}`.`ID` = `{$alias_event}`.`post_id`
+					WHERE
+						`{$alias_event}`.`meta_value` = `{$wpdb->posts}`.`ID`
+						AND `{$alias_event}`.`meta_key` IN ( {$event_meta_keys} )
+						AND `{$alias_attendee}`.`post_type` IN ( {$attendee_types} )
+						AND `{$alias_attendee}`.`ID` IN ( {$attendee_ids} )
+					LIMIT 1
+				)
+			" );
+	}
```

Comment:
:no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: 1b9e1197847de323244257d0b508dbb0</sub>